src/main/java/frc/robot/commands/ramp/ResetRamp.java

```diff
@@ -41,13 +43,21 @@ public void execute() {}
   @Override
   public void end(boolean interrupted) {
     ramp.stopMotor();
-    ramp.resetEncoder();
+    if(ramp.getReversedSwitchState()) ramp.resetEncoder();
   }
 
   // Returns true when the command should end.
   @Override
   public boolean isFinished() {
-    return ((ramp.getReversedSwitchState()) || ((Timer.getFPGATimestamp() - startTime) >= 5)); //Assuming that Forward Switch is the top one
+    if (ramp.getReversedSwitchState()) {
+      ramp.stopMotor();
+    if(rampStopTime.equals(null)) rampStopTime = Timer.getFPGATimestamp();
```

Comment:
Please format this correctly with braces. We have enough different levels of programmers that this style where the braces are left-off will be highly prone to error.